test/unit/lodash.spec.js

```diff
@@ -0,0 +1,568 @@
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length.within(1, 2);
+            if (params[1] !== undefined && params.length === 2) {
+                expect(params[1]).to.be.an('object');
+            }
+        });
+        it('should check if second argument have validate properties', function () {
+            var options = params[1];
+            if (options !== undefined && params.length === 2) {
+                expect(options).to.have.any.keys('length', 'omission', 'separator');
+            }
+            // powinien miec max 3 wlasciowsci
+            // pierwsza number
+        });
+        it('should return truncated string', function () {
+                expect(_.truncate.apply(_, params)).to.eql('some very long text...');
+            }
+        );
+    });
+
+    describe('unescape', function () {
+        before(function () {
+            params = datasets.unescape();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.unescape.apply(_, params)).to.eql('tom & jerry were old friends');
+            }
+        );
+    });
+
+    describe('upperCase', function () {
+        before(function () {
+            params = datasets.upperCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.upperCase.apply(_, params)).to.eql('SOME STRING TO TEST FUNCTION');
+            }
+        );
+    });
+
+    describe('upperFirst', function () {
+        before(function () {
+            params = datasets.upperFirst();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.upperFirst.apply(_, params)).to.eql('Some string to test function');
+            }
+        );
+    });
+
+    describe('words', function () {
+        describe('words1', function () {
+            before(function () {
+                params = datasets.words1();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params).to.have.length.within(1, 2);
+                if (params[1] !== undefined && params.length === 2) {
+                    expect(params[1]).to.be.a('string');
+                }
+            });
+            it('should return words of string', function () {
+                    expect(_.words.apply(_, params)).to.eql([ 'bread', 'butter', 'milk' ]);
+                }
+            );
+        });
+        describe('words2', function () {
```

Comment:
Both I can solve without RegExp parameter. And what is the difference between words1 and words2?